skyllh/core/expectation_maximization.py

```diff
@@ -81,73 +82,23 @@ def maximization_em(e_sig, t):
     sigma = [max(1, s) for s in sigma]
 
     return mu, sigma, ns
+   
 
+class ExpectationMaximizationTimefit():
+    """ Helper class for using expectation maximization for fitting the flare time
+    """
 
-class ExpectationMaximizationUtils(object):
     def __init__(self, ana):
-        """Creates a expectation maximization utility class for time-dependent
-        single dataset point-like source analysis assuming a single source.
-
+        """
         Parameters
         ----------
-        ana : instance of TimeIntegratedMultiDatasetSingleSourceAnalysis
-            Analysis instance which will be stored in this class.
-        """
-        self.ana = ana
+        ana : analysis instance
 
-    @property
-    def ana(self):
-        """The TimeIntegratedMultiDatasetSingleSourceAnalysis instance.
         """
-        return self._ana
-
-    @ana.setter
-    def ana(self, analysis):
-        if not isinstance(analysis, TimeIntegratedMultiDatasetSingleSourceAnalysis):
-            raise TypeError("The ana argument must be an instance of "
-                            "'TimeIntegratedMultiDatasetSingleSourceAnalysis'.")
-        self._ana = analysis
-
+        self.ana = ana
 
-    def change_time_pdf(self, gauss=None, box=None):
-        """Changes the time pdf.
 
-        Parameters
-        ----------
-        gauss : dict | None
-            None or dictionary with {"mu": float, "sigma": float}.
-        box : dict | None
-            None or dictionary with {"start": float, "end": float}.
-        """
-        ana = self.ana
-
-        if gauss is None and box is None:
-            raise TypeError("Either gauss or box have to be specified as time pdf.")
-
-        grl = ana._data_list[0].grl
-        # redo this in case the background pdf was not calculated before
-        time_bkgpdf = BackgroundUniformTimePDF(grl)
-        if gauss is not None:
-            time_sigpdf = SignalGaussTimePDF(grl, gauss['mu'], gauss['sigma'])
-        elif box is not None:
-            time_sigpdf = SignalBoxTimePDF(grl, box["start"], box["end"])
-
-        time_pdfratio = SigOverBkgPDFRatio(
-            sig_pdf=time_sigpdf,
-            bkg_pdf=time_bkgpdf,
-            pdf_type=TimePDF
-        )
-
-        # the next line seems to make no difference in the llh evaluation. We keep it for consistency
-        ana._llhratio.llhratio_list[0].pdfratio_list[2] = time_pdfratio
-        # this line here is relevant for the llh evaluation
-        ana._llhratio.llhratio_list[0]._pdfratioarray._pdfratio_list[2] = time_pdfratio
-
-        #  change detector signal yield with flare livetime in sample (1 / grl_norm in pdf),
-        #  rebuild the histograms if it is changed...
-        #  signal injection?
-
-    def get_energy_spatial_signal_over_backround(self, fitparams):
+    def get_energy_spatial_signal_over_background(self, fitparams):
```

Comment:
Isn't this method also user-analysis dependent and should live within the file which contains the `create_analysis` function?